daml-lf/spec/daml-lf-1.rst

```diff
@@ -1955,187 +1975,326 @@ exact output.
 
 ::
 
-  Evaluation environment
-    E ::= ε                                         -- EnvEmpty
-       |  Mod:W ↦ v · E                             -- EnvVal
-
   Evaluation result
     r ::= Ok v                                      -- ResOk
        |  Err t                                     -- ResErr
```

Comment:
In the long run we probably need to distinguish different error classes, such as errors raised via `error`, arithmetic errors, pattern match failures, etc. I would suggest making that the subject of a separate PR though.

Reply:
I think that can be done easily enough by giving `t` some structure, without really changing most of the semantics.